lib/fluent/plugin/out_google_cloud.rb

```diff
@@ -437,10 +437,15 @@ def write(chunk)
 
           ts_secs, ts_nanos = compute_timestamp(record, time)
           if @use_grpc
-            entry.metadata.timestamp = Google::Protobuf::Timestamp.new(
-              seconds: ts_secs,
-              nanos: ts_nanos
-            )
+            # If "seconds" or "nanos" are not integers (e.g. null), we will omit
+            # the timestamp field and defer the decision of whether to set it to
+            # the current timestamp to the downstream Logging API.
+            if ts_secs.is_a?(Integer) && ts_nanos.is_a?(Integer)
```

Comment:
This seems to be a much broader check than what you're testing below. Shouldn't you also check non-integer values of the timestamp? Should you check when only one of `seconds` or `nanos` is `nil`?